csharp/AlgorithmWorkshop/QuickSort.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace AlgorithmWorkshop
+{
+    public static class QuickSort
+    {
+        public static List<double> Sort(List<double> listToSort)
+        {
+            if (listToSort.Count <= 1) return listToSort;
+            var partition = Partition(listToSort);
+            var sorted = new List<double>();
+            sorted.AddRange(Sort(partition[0]));
+            sorted.AddRange(partition[1]);
+            sorted.AddRange(Sort(partition[2]));
+
+            return sorted;
+        }
+
+        public static List<List<double>> Partition(List<double> listToSort, int pivotIndex = 0)
+        {
+            if (!listToSort.Any()) throw new Exception("List cannot be empty.");
+            if (pivotIndex == 0) pivotIndex = new Random().Next(listToSort.Count - 1);
```

Comment:
What if I wanted pivotIndex to equal 0? Is there a way to both support specifying a pivot of 0 _and_ using a random index by default?